typedlua/tlchecker.lua

```diff
@@ -1962,14 +1962,23 @@ function tlchecker.typecheck (ast, subject, filename, strict, integer, color)
   return env.messages
 end
 
-local function get_source_line(filename, l)
+local function get_source_line(subject, filename, l)
+  if not subject then
+    local file = assert(io.open(filename))
+    subject = file:read("*a")
+    file:close()
+  end
   local i = 1
-  for source_line in io.lines(filename) do
+  local lstart = 1
+  repeat
+    local lend, lnext = subject:find("\r?\n", lstart)
     if i == l then
-      return (string.gsub(source_line, "\t", " "))
+      local line = subject:sub(lstart, lend and lend-1 or #subject)
+      return (line:gsub("\t", " "))
```

Comment:
(previous comment dropped off): escape or strip all non-printable characters

Reply:
I saw the comment, I think this was done as a hack to fix placement of the arrow when tabs are involved, but that placement will already be broken in several other situations, such as with any line that has multibyte characters.